src/main/java/com/checkmarx/flow/controller/ADOController.java

```diff
@@ -50,86 +45,60 @@
-            @RequestParam(value = "preset", required = false) String preset,
-            @RequestParam(value = "incremental", required = false) Boolean incremental,
-            @RequestParam(value = "exclude-files", required = false) List<String> excludeFiles,
-            @RequestParam(value = "exclude-folders", required = false) List<String> excludeFolders,
-            @RequestParam(value = "override", required = false) String override,
-            @RequestParam(value = "bug", required = false) String bug,
-            @RequestParam(value = "ado-issue", required = false) String adoIssueType,
-            @RequestParam(value = "ado-body", required = false) String adoIssueBody,
-            @RequestParam(value = "ado-opened", required = false) String adoOpenedState,
-            @RequestParam(value = "ado-closed", required = false) String adoClosedState,
-            @RequestParam(value = "app-only", required = false) Boolean appOnlyTracking
-    ){
+            ControllerRequest controllerRequest,
+            AdoDetailsRequest adoDetailsRequest
+    ) {
         String uid = helperService.getShortUid();
         MDC.put("cx", uid);
         log.info("Processing Azure PULL request");
         validateBasicAuth(auth);
+        controllerRequest = ensureNotNull(controllerRequest);
+        adoDetailsRequest = ensureDetailsNotNull(adoDetailsRequest);
 
-        if(!PULL_EVENT.contains(body.getEventType()) || !body.getResource().getStatus().equals("active")){
+        if (!PULL_EVENT.contains(body.getEventType()) || !body.getResource().getStatus().equals("active")) {
             log.info("Pull requested not processed.  Event was not opened ({})", body.getEventType());
             return ResponseEntity.status(HttpStatus.OK).body(EventResponse.builder()
                     .message("No processing occurred for updates to Pull Request")
                     .success(true)
                     .build());
         }
 
-        FlowOverride o = ScanUtils.getMachinaOverride(override);
+        FlowOverride o = ScanUtils.getMachinaOverride(controllerRequest.getOverride());
 
         try {
             Resource resource = body.getResource();
             Repository repository = resource.getRepository();
             String pullUrl = resource.getUrl();
             String app = repository.getName();
 
-            if(repository.getName().startsWith(properties.getTestRepository())){
+            if (repository.getName().startsWith(properties.getTestRepository())) {
                 log.info("Handling ADO Test Event");
                 return ResponseEntity.status(HttpStatus.OK).body(EventResponse.builder()
                         .message("Test Event").success(true).build());
             }
 
-            if(!ScanUtils.empty(application)){
-                app = application;
+            if (StringUtils.isNotEmpty(controllerRequest.getApplication())) {
+                app = controllerRequest.getApplication();
```

Comment:
you are calculating this value 2 times. use optional to calculate it a single time

Reply:
After a short investigation, using `Optional` vs `if` seems more like a matter of taste to me. Here are some findings:
https://codeburst.io/replace-your-if-else-if-nested-if-not-null-then-get-blocks-with-java-8-optional-s-ed9695a6331e
https://stackoverflow.com/a/56235329
https://codereview.stackexchange.com/a/223329

As for calling getters twice - do you mean the issue of performance? This seems like a micro-optimization to me. One may also argue that the creation of Optional is an expensive operation.